lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -612,39 +701,304 @@ void ScienceSensorsSystemPrivate::GenerateOctrees()
   for (int i = 0; i < this->timeSpaceCoords.size(); ++i)
   {
     this->spatialOctrees.push_back(
-      pcl::octree::OctreePointCloudSearch<pcl::PointXYZ>(this->spatialRes));
+      pcl::octree::OctreePointCloudSearch<pcl::PointXYZ>::Ptr(
+        new pcl::octree::OctreePointCloudSearch<pcl::PointXYZ>(
+          this->spatialRes)));
 
     // Populate octree with spatial coordinates
-    this->spatialOctrees[i].setInputCloud(this->timeSpaceCoords[i]);
-    this->spatialOctrees[i].addPointsFromInputCloud();
+    this->spatialOctrees[i]->setInputCloud(this->timeSpaceCoords[i]);
+    this->spatialOctrees[i]->addPointsFromInputCloud();
   }
 }
 
 /////////////////////////////////////////////////
-void ScienceSensorsSystemPrivate::PublishData()
+bool ScienceSensorsSystemPrivate::comparePclPoints(
+  const pcl::PointXYZ &a,
+  const pcl::PointXYZ &b)
 {
-  this->cloudPub.Publish(this->PointCloudMsg());
-  this->tempPub.Publish(this->tempMsg);
-  this->chlorPub.Publish(this->chlorMsg);
-  this->salPub.Publish(this->salMsg);
+  // Comparison between points is arbitrary. This function is only used for
+  // set operations, not for literal sorting.
+  if (a.x < b.x)
```

Comment:
This could be done in one line 
```suggestion
  return a.x < b.x && a.y < b.y && a.z < b.z
```

removing the rest of the lines.

Reply:
e2efd70